### ISSUE TITLE
LIBPATH should be double backslashes within quoted or single backslash without quoted or forward slash

### ISSUE THREAD

**Issue by Domain** [fastbuild/fastbuild#467]
My project goes well with 0.95. When I upgrade to 0.96, the linker complains some lib files not found. I run the command in verbose mode, and the only difference is that, the forward slash is replaced with back slash:
0.95: /LIBPATH:"lib3rd/lib/"  -->  0.96: /LIBPATH:"lib3rd\lib\"
I change the back slash to forward slash, or to double back slash, and all goes well again.
This confused me. I thought linker support single back slash in path.

**Comment by Domain** [comment]
It goes well when I removing the double quote!
/LIBPATH:lib3rd\lib\

**Comment by Domain** [comment]
Oops! I am sorry, this is my mistake. I can change my bff file to remove the double quote

**Comment by ffulin** [comment]
The ms tools treat a backslash as an escape characters within a quoted string, but only for certain characters (like " and \, while the first letter of the subdir "lib" is not escaped and it retains the slash). This causes a lot of confusion, especially at the end of paths, when you end up breaking termination of the string and end up with lots of super weird errors.

FASTBuild tried to detect this case and auto-fix include paths to remove final slashes. It should probably do the same for LIBPATH dirs.

Having said all this, I'd like to understand why there was a behaviour change for LIBPATH in v0.96. Can you please give me some information about how the LIBPATH is being populated in your bff?

**Comment by ffulin** [comment]
I investigated this, and while trailing backslashes have always been a problem in quoted LIBPATH strings, in v0.96 FASTBuild started replacing the slash direction in these paths. This made previously harmless trailing forward slashes be converted into the dangerous trailing backslashes.

I have added code which removes any trailing backslashes in LIBPATHs that would escape the closing quote, regardless of whether they were put there by the user or added by FASTBuild: https://github.com/fastbuild/fastbuild/commit/5edf4c87ffec257eadb8348bc7e50fccb6546150. This fix will be in the next release (v0.97). I'll leave this issue open until that version is released.

To avoid this problem until that version is released, I recommend removing the trailing slash from any LIBPATHs.
